homeassistant/components/sensor/rmvtransport.py

```diff
@@ -0,0 +1,202 @@
+                'minutes': self.data.departures[0].get('minutes'),
+                'departure_time':
+                    self.data.departures[0].get('departure_time'),
+                'product': self.data.departures[0].get('product'),
+            }
+        except IndexError:
+            return {}
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return "min"
+
+    def update(self):
+        """Get the latest data and update the state."""
+        self.data.update()
+        if not self.data.departures:
+            self._state = None
+            self._icon = ICONS[None]
+            return
+        if self._name == DEFAULT_NAME:
+            self._name = self.data.station
+        self._station = self.data.station
+        self._state = self.data.departures[0].get('minutes')
+        self._icon = ICONS[self.data.departures[0].get('product')]
+
+
+class RMVDepartureData:
+    """Pull data from the opendata.rmv.de web page."""
+
+    def __init__(self, station_id, destinations, directions,
+                 lines, products, time_offset, max_journeys):
+        """Initialize the sensor."""
+        import RMVtransport
+        self.station = None
+        self._station_id = station_id
+        self._destinations = destinations
+        self._directions = directions
+        self._lines = lines
+        self._products = products
+        self._time_offset = time_offset
+        self._max_journeys = max_journeys
+        self.rmv = RMVtransport.RMVtransport()
+        self.departures = {}
+
+    def update(self):
+        """Update the connection data."""
+        try:
+            _data = self.rmv.get_departures(self._station_id,
+                                            products=self._products,
+                                            maxJourneys=50)
+        except ValueError:
+            self.departures = {}
+            _LOGGER.warning("Returned data not understood")
+            return
+        self.station = _data.get('station')
+        _deps = []
+        for journey in _data['journeys']:
+            # find the first departure meeting the criteria
+            _nextdep = {ATTR_ATTRIBUTION: ATTRIBUTION}
+            if self._destinations:
+                dest_found = False
+                for dest in self._destinations:
+                    if dest in journey['stops']:
+                        dest_found = True
+                        _nextdep['destination'] = dest
+                if not dest_found:
+                    continue
+            elif self._lines and journey['number'] not in self._lines:
+                continue
+            elif journey['minutes'] < self._time_offset:
+                continue
+            for k in ['direction', 'departure_time', 'product', 'minutes']:
```

Comment:
`k` doesn't explain what the variable is. Please use a descriptive variable name, eg `attr`.